src/index.d.ts

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import * as monacoEditor from 'monaco-editor';
+
+declare class MonacoEditor extends React.Component<MonacoEditorProps> {
+  constructor(props: Partial<MonacoEditorProps>);
+
+  componentDidMount(): void;
+  componentDidUpdate(prevProps: Partial<MonacoEditorProps>): void;
+  componentWillUnmount(): void;
+
+  editorDidMount(editor: monacoEditor.editor.IEditor, monaco: typeof monacoEditor): void;
+  editorWillMount(monaco: any): void;
+
+  afterViewInit(): void;
+  initMonaco(): void;
+  destroyMonaco(): void;
```

Comment:
I think these methods are private, not the instance methods used by users. So we should omit them.

Reply:
This is what I meant in this comment: https://github.com/superRaytin/react-monaco-editor/issues/114#issuecomment-386413425